lib/from_erlang.ml

```diff
@@ -31,13 +39,26 @@ let rec expr_of_erlang_expr = function
     | F.ClsCase (_, _, _, _) | F.ClsFun (_, _, _, _) -> failwith "not implemented"
     ) in
     Case (expr_of_erlang_expr e, cs)
+  | ExprLocalFunRef (_line_t, name, arity) ->
+     (* TODO: support local `fun F/A` *)
+     failwith "not implemented"
+  | ExprRemoteFunRef (_line_t, m, f, a) ->
+     MFA (expr_of_atom_or_var m, expr_of_atom_or_var f, expr_of_integer_or_var a)
+  | ExprFun (_line_t, name, clauses) ->
+     (* TODO: fun _(...) -> ... end *)
+     failwith "not implemented"
   | ExprLocalCall (_line_t, f, args) ->
      App (expr_of_erlang_expr f, List.map ~f:expr_of_erlang_expr args)
   | ExprRemoteCall (_line_t, _line_m, m, f, args) ->
      let mfa = MFA (expr_of_erlang_expr m, expr_of_erlang_expr f, Constant (Number (List.length args))) in
      App (mfa, List.map ~f:expr_of_erlang_expr args)
+  | ExprMatch (line_t, pat, e) ->
+     (* HACK: `A = B` convert to case expr `case B of A -> B end` *)
+     expr_of_erlang_expr (F.ExprCase (line_t, e, [F.ClsCase (line_t, pat, None, e)]))
```

Comment:
## It's Ninja Pattern Matching. 

Can the following expressions see the bound variables in pattern `A` ?

For example,


```erlang
{Name, Age} = foo(),
<expr>
```

is equals to:

```erlang
case foo() of
  {Name, Age} ->
    <expr>
end
```

but not equals to:
```erlang
case foo() of
  {Name, Age} ->
    foo()
end,
<expr>
```

Reply:
You're right!
Then I think it is needed to add some extensions into `Ast_intf.expr`.
I'll create issue and replace the code with `NotImplemented` error.

Reply:
https://github.com/dwango/fialyzer/pull/78/commits/0e84ef82149c15a23a3da4203d5092029db76693